clerk/middleware.go

```diff
@@ -30,3 +31,21 @@ func addSessionToContext(request *http.Request, session *Session) *http.Request
 	updated := context.WithValue(ctx, ActiveSession, session)
 	return request.WithContext(updated)
 }
+
+func WithNetworkless(client Client) func(handler http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			token := r.Header.Get("Authorization")
+
+			claims, err := client.VerifyToken(token)
+			if err != nil {
+				w.WriteHeader(http.StatusUnauthorized)
+				_, _ = w.Write([]byte(err.Error()))
+				return
+			}
+
+			ctx := context.WithValue(r.Context(), ActiveClaims, claims)
+			next.ServeHTTP(w, r.WithContext(ctx))
+		})
+	}
```

Comment:
Apart from the `Authorization` header, we additionally have to try and fetch the token from the `__session` cookie, based on architecture no. 1, right?

Reply:
Good catch, it's possible someone to serve a UI via a Go binary